Sources/SourceKitBazelBSP/RequestHandlers/TargetSourcesHandler.swift

```diff
@@ -62,10 +61,40 @@ final class TargetSourcesHandler {
         return BuildTargetSourcesResponse(items: srcs)
     }
 
+    private func computeCopyDestinations(for src: URI) -> [DocumentURI]? {
+        guard let srcPath = src.fileURL?.path else {
+            return nil
+        }
+
+        var destinations: [DocumentURI] = []
+
+        let workspaceRoot = initializedConfig.rootUri
+        let execRoot = initializedConfig.executionRoot
+
+        func trimmedRelativePath(fullPath: String, base: String) -> String {
+            var relativePath = fullPath.dropFirst(base.count)
+            if relativePath.first == "/" {
+                relativePath = relativePath.dropFirst()
+            }
+            return String(relativePath)
+        }
+
+        if srcPath.hasPrefix(workspaceRoot) {
+            // Workspace file -> single execroot location Bazel copies to.
+            // We need to tell the LSP about the path of each file in execution root to later help it mapping it again to original Workspace path.
+            let relativePath = trimmedRelativePath(fullPath: srcPath, base: workspaceRoot)
+            let destination = DocumentURI(filePath: execRoot + "/" + relativePath, isDirectory: true)
```

Comment:
Is it correct for isDirectory to be `true` here? This is the path to the source files, no?

Reply:
Yup, it should be fine without it `isDirectory` as well